`ContractTxInfo` construction into helper functions in Signature Request Methods

wthrajat opened the pull request:
This PR refactors the construction of `ContractTxInfoForSender` and `ContractTxInfoForRecvr` into separate helper functions:

- [x] `build_contract_tx_info_for_sender`: Handles the construction of the `txs_info` vector in the `req_sigs_for_sender_once` function. This encapsulates the logic for preparing the sender's contract transaction information, reducing clutter in the main function.

- [x] `build_contract_tx_info_for_recvr`: Constructs the `txs` vector in the `req_sigs_for_recvr_once` function. Simplifies the code in the receiver's signature request method, making it easier to understand and maintain.

TLDR:

Solves 2 TODOs:
1. Take this construction out of function body.
2. Take the message construction out of the function body.

Shourya742 commented:
@wthrajat Can you squash the commits?

mojoX911 commented:
Nack. 
There is no benefit of taking these one liners out in their own function. 

This is in general a bad idea for maintainability.

wthrajat commented on src/taker/routines.rs:
```diff
@@ -92,23 +92,12 @@ pub fn handshake_maker(socket: &mut TcpStream) -> Result<(), TakerError> {
-        socket.peer_addr()?
-    );
-
-    // TODO: Take this construction out of function body.
```
@mojoX911 This PR solves 2 TODOs

mojoX911 replied:
The intention of this TODO is to take the message construction out of the `send_message` function body to increase readability. The TODO is already solved here in this function, but for some reason, I forgot to remove it. So it's redundant. 

It still applies in the `req_sigs_for_recvr_once` function where the message is constructed within the `send_message` function. That could be a valid PR.

mojoX911 commented:
@wthrajat if you can just address this https://github.com/citadel-tech/coinswap/pull/260#discussion_r1754930893 then this is good for merge.

Instead of taking the message construction in a new function, keep them in the old call site itself and move them out from the `send` function.

wthrajat commented:
@mojoX911 I’ve removed the redundant TODO in `req_sigs_for_sender_once()` and moved the message construction outside of the `send_message()` call in `req_sigs_for_recvr_once()`.

The TODO comment was a bit confusing as I thought it referred to moving the construction outside the parent functions rather than the `send_message()` calls.
